src/figuresClasses.ts

```diff
@@ -1,11 +1,85 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('a or b or c does not have valid value');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('a or b or c does not have valid value');
```

Comment:
[CODE STYLE] - The error message here should also be more descriptive. Instead of stating 'a or b or c does not have valid value', it could indicate that the given side lengths do not form a valid triangle.